README.md

```diff
@@ -181,6 +184,125 @@ jobs:
+does not require Docker. The composite action, however, requires a Python3 environment to be setup
+on the action runner. All GitHub-hosted runners (Ubuntu, Windows Server and macOS) provide a suitable
+Python3 environment out-of-the-box.
+
+Self-hosted runners may require setting up a Python environment first:
+
+```yaml
+- name: Setup Python
+  uses: actions/setup-python@v2
+  with:
+    python-version: 3.8
+```
+
+### Isolating composite action from your workflow
+
+Note that the composite action modifies this Python environment by upgrading PIP and installing dependency packages.
+If this conflicts with actions that later run Python in the same workflow (rare case), it is recommended to run this action in an isolated workflow.
+This is similar to the workflows shown in [Use with matrix strategy](#use-with-matrix-strategy).
+
+Your CI workflow should upload all test result XML files:
+
+```yaml
+build-and-test:
+   name: "Build and Test"
+   runs-on: macos-latest
+
+   steps:
+   - …
+   - name: Upload Unit Test Results
+     if: always()
+     uses: actions/upload-artifact@v2
+     with:
+       name: Unit Test Results
+       path: test-results/**/*.xml
+```
+
+Your dedicated publish-unit-test-result-workflow then downloads these files and runs the action there:
+
+```yaml
+publish-test-results:
+ name: "Publish Unit Tests Results"
+ needs: build-and-test
+ runs-on: macos-latest
+ # the build-and-test job might be skipped, we don't need to run this job then
+ if: success() || failure()
+
+ steps:
+   - name: Download Artifacts
+     uses: actions/download-artifact@v2
+     with:
+       path: artifacts
+
+   - name: Publish Unit Test Results
+     uses: EnricoMi/publish-unit-test-result-action/composite@v1
+     with:
+       files: artifacts/**/*.xml
+```
+
+### Slow startup of composite action
+
+In some environments, the composite action startup can be slow due to installing Python dependencies.
+This is usually the case for **Windows** runners (in this example 35 seconds startup time):
+
+```
+Mon, 03 May 2021 11:57:00 GMT   ⏵ Run ./composite
+Mon, 03 May 2021 11:57:00 GMT   ⏵ Check for Python3
+Mon, 03 May 2021 11:57:00 GMT   ⏵ Install Python dependencies
+Mon, 03 May 2021 11:57:35 GMT   ⏵ Publish Unit Test Results
+```
+
+This can be improved by caching the PIP cache directory. If you see the following warning in
+the composite action output, then installing the `wheel` package can also be beneficial (see further below):
+
+```
+Using legacy 'setup.py install' for …, since package 'wheel' is not installed.
+```
+
+You can [cache files downloaded and built by PIP](https://github.com/actions/cache/blob/main/examples.md#python---pip)
+using the `actions/cache` action, and conditionally install the `wheel`package as follows:
+
+```yaml
+- name: Cache PIP Packages
+  uses: actions/cache@v2
+  id: cache
+  with:
+    path: ~\AppData\Local\pip\Cache
```

Comment:
This should probably be `${{ env.LOCALAPPDATA }}\pip\Cache` to be robust. For matrix strategy invovling different OS the cache directory should probably be set as an `include` in the matrix definition.

Reply:
The `env.LOCALAPPDATA` does not seem to work:

https://github.com/EnricoMi/publish-unit-test-result-action/blob/5d3c5978459e7f29423e1dfee3435d7cde1c0c36/.github/workflows/ci-cd.yml#L199-L206

https://github.com/EnricoMi/publish-unit-test-result-action/runs/2504086460?check_suite_focus=true#step:4:1

Reply:
This is probably a variable that is not known to GitHub Actions but exists for the script once it runs on the runner.

Reply:
LOCALAPPDATA exists on the Windows machine, but GitHubs ${{ env }} does not know about it: https://github.com/EnricoMi/publish-unit-test-result-action/runs/2506880581?check_suite_focus=true#step:4:79

Reply:
I'll stick to the paths recommended in https://github.com/actions/cache/blob/main/examples.md#python---pip, also to keep this example as simple as possible. User's can improve and adjust this to their needs.

Reply:
Hm, that is really strange. Why would they not include that? Or does `${{ env }}` only contain the environment variables _set_ by the workflow and not the discovered ones? Sorry I sent you on a goose chase.

Reply:
No worries, I learnt something.